apps/rush-lib/src/api/ChangeFile.ts

```diff
@@ -81,7 +81,7 @@ export class ChangeFile {
   public generatePath(): string {
     let branch: string | undefined = undefined;
     try {
-      branch = gitInfo().branch;
+      branch = Git.getGitInfo()!.branch;
```

Comment:
>!.branch [](start = 31, length = 8)

How do we know that `getGitInfo()` doesn't return undefined here?

Reply:
It could, but the try/catch would catch it.

Reply:
Agreed.  A null reference exception is always a program malfunction.  We would never intentionally catch this sort of exception.

---
In reply to: [241253975](https://github.com/Microsoft/web-build-tools/pull/994#discussion_r241253975) [](ancestors = 241253975)